source/mod_dist.f90

```diff
@@ -290,58 +850,41 @@ subroutine dist_finaliseDist
   write(lout,"(a,2(1x,i0),1x,f15.7,2(1x,i0))") "DIST> Reference particle species [A,Z,M,Q,ID]:", aa0, zz0, nucm0, qq0, pdgid0
   write(lout,"(a,1x,f15.7)")       "DIST> Reference energy [Z TeV]:", c1m6*e0/qq0
 
-  do j=napx+1,npart
-    partID(j)   = j
-    parentID(j) = j
-    pstop(j)    = .true.
-    ejv(j)      = zero
-    dpsv(j)     = zero
-    oidpsv(j)   = one
-    mtc(j)      = one
-    naa(j)      = aa0
-    nzz(j)      = zz0
-    nqq(j)      = qq0
-    pdgid(j)    = pdgid0
-    nucm(j)     = nucm0
-    moidpsv(j)  = one
-    omoidpsv(j) = zero
-  end do
-
 end subroutine dist_finaliseDist
 
 ! ================================================================================================ !
 !  A. Mereghetti and D. Sinuela Pastor, for the FLUKA Team
 !  V.K. Berglyd Olsen, BE-ABP-HSS
-!  Last modified: 2018-10-30
+!  Updated: 2019-07-08
 ! ================================================================================================ !
 subroutine dist_echoDist
 
   use mod_common
   use mod_common_main
-  use mod_units, only : f_open, f_close
+  use mod_units
 
-  integer j
+  integer j, fUnit
   logical cErr
 
-  call f_open(unit=dist_echoUnit,file=dist_echoFile,mode='w',err=cErr,formatted=.true.)
+  call f_requestUnit(dist_echoFile, fUnit)
+  call f_open(unit=fUnit,file=dist_echoFile,mode='w',err=cErr,formatted=.true.)
   if(cErr) goto 19
 
-  rewind(dist_echoUnit)
-  write(dist_echoUnit,"(a,1pe25.18)") "# Total energy of synch part [MeV]: ",e0
-  write(dist_echoUnit,"(a,1pe25.18)") "# Momentum of synch part [MeV/c]:   ",e0f
-  write(dist_echoUnit,"(a)")          "#"
-  write(dist_echoUnit,"(a)")          "# x[mm], y[mm], xp[mrad], yp[mrad], sigmv[mm], ejfv[MeV/c]"
+  rewind(fUnit)
+  write(fUnit,"(a,1pe25.18)") "# Total energy of synch part [MeV]: ",e0
```

Comment:
I would replace "synch" with "reference"

Reply:
Sure. I can review the echo file in general. The 1pe25.18 format is also a decimal too many.